fathom/src/driver.rs

```diff
@@ -340,4 +344,54 @@ impl<'surface, 'core> Driver<'surface, 'core> {
 
         is_ok
     }
+
+    fn report_read_error(&self, err: binary::ReadError) {
+        // Use the currently set codespan configuration
+        let term_config = self.codespan_config.clone();
+
+        let diagnostic = Diagnostic::from(err);
+        let mut writer = BufferedStandardStream::stderr(if atty::is(atty::Stream::Stderr) {
+            ColorChoice::Auto
+        } else {
+            ColorChoice::Never
+        });
+        let dummy_files = SimpleFiles::<String, String>::new();
```

Comment:
I think I use the existing `diagnostic_writer` and file DB in #338. Not sure what is best, but maybe we can get this merged first as it has the friendlier errors! 👍